index.js

```diff
@@ -195,21 +187,16 @@ let Fixtures = {
      * @return {Promise} The ChakramResponse from the creation of the records and/or links
      */
     create(models, options = {}) {
-        if (_.isUndefined(this._headers['OAuth-Token'])) {
-            if (++this._sessionAttempt > this._maxSessionAttempts) {
-                throw new Error('Max number of login attempts exceeded!');
-            }
-
-            return this._adminLogin().then(() => this.create(models, options));
+        if (this._needsLogin()) {
+            return this._adminLogin().then(() => this.create(models, options)).catch((response) => {
+                throw response;
+            });
         }
 
         if (!_.isArray(models)) {
             models = [models];
         }
 
-        // reset `_sessionAttempt`
-        this._sessionAttempt = 0;
```

Comment:
I've moved the complexity of dealing with retries to `utils._oauthRequest` where it can be reused. Fixtures is some of our most complex code and this will make it more focused on dealing with the mechanics of record manipulation.